Suppress accent from phrase on Broker endpoint

mickbird opened the pull request:
Hi everyone,

For french speakers, it can be unreadable to implement complexe regex matching for broker due to accentued characters (for example, the action "play" will be "regard(e|é)".
This PR implements the phrase normalization by removing the accents to convert "regardé" --> "regarde".

keydon commented:
Thank you for your time and effort. Though I cannot merge it right now, since there are some issues to be resolved.

1. The tests need to be adjusted accordingly in this PR. See the error output of `npm test`
2. btw. the library you choose does not only kill accents, it also kills german umlauts. Stripping umlauts is not desired in german.
So maybe only apply stripping accents for the frensh broker only for now :)
3. the accents would not only be stripped from the command-parts, butalso be stripped i.e. from the movie titles. Although I guess the fuzzy search would still find the correct movie. A confirmation that you tested it with movies with accents in their title would be great.

mickbird commented:
Dear Lukas,

Thanks for your feedback.
1. I left the test cases untouched and modified the code to prevent breaking changes.

2. A new implementation has been made with a globalConfig variable `brokerAccentInsensitiveMatch` which controls pattern matching with accents.
    If set to `false` (default), `e` and `ë` are not equal.
    If set to `true`, `e`, `é`, `ê`, `è` and `ë` are equal (accents are converted)
   The associated environment variable `BROKER_ACCENT_INSENSITIVE_MATCH` can be used to configure the behaviour.

3. I tested that removing accents from the movie title doesn't affect fuzzy search. The following requests have been used :
```sh
   curl --location --request POST 'YOUR_NODE_SERVER/broker?phrase=démarre 12 Angry Men&lang=fr' --header 'Content-Type: application/json' --data-raw '{"token":"YOUR_CONNECTION_PASSWORD"}'
   curl --location --request POST 'YOUR_NODE_SERVER/broker?phrase=démarre 12 hommes en colere&lang=fr' --header 'Content-Type: application/json' --data-raw '{"token":"YOUR_CONNECTION_PASSWORD"}'
   curl --location --request POST 'YOUR_NODE_SERVER/broker?phrase=démarre 12 hommes en colère&lang=fr' --header 'Content-Type: application/json' --data-raw '{"token":"YOUR_CONNECTION_PASSWORD"}'
```

During dev, I found tricky that the language file caching cannot be disabled. When writing rules, it would be easy to modify the json file and send a new request without restarting the server/changing the language.
   I added a globalConfig variable `brokerLanguageCacheEnable` which control how and when the language file is loaded.
    If set to `true` (default), the language file is loaded when the previous and actual requests are different.
    If set to `false`, the language file is reloaded on every request.
   The associated environment variable `BROKER_LANGUAGE_CACHE_ENABLE` can be used to configure the behaviour.

The `README.MD` file has been partially updated to reflect the proposed changes.

mickbird commented:
Dear Lukas,

Thanks for merging the PR.
I saw the Github Action ran without issue. However, the Docker image has not been updated. (build and push tasks check if the repository name is not yours.
Is it voluntary ? If so, how is it planned to do the release ?
Best regards

keydon commented:
ah sorry, I forgot. It is released now.

The issue is, I dont have enough permissions on omertus repository, so I am not able to configure any secrets here.
I have to merge to my repository in order for github-action to be able to build&push it to docker-hub.
I plan to automate that, but did not do it yet ;D
